execution_engine/src/core/engine_state/mod.rs

```diff
@@ -157,6 +157,16 @@ impl EngineState<LmdbGlobalState> {
             .put_stored_values(CorrelationId::new(), state_root_hash, stored_values)
             .map_err(Into::into)
     }
+
+    /// Writes state cached in an EngineState<ScratchEngineState> to LMDB.
+    #[deprecated(since = "2.0.0", note = "renamed to `write_scratch_to_lmdb`")]
+    pub fn write_scratch_to_db(
```

Comment:
I think rather than re-adding this as a deprecated function, we should just rename `write_scratch_to_lmdb` to `write_scratch_to_db` since that is how we'll be naming things as we go forward, moving away from LMDB.

Reply:
Wasn't the whole point to not "remove" the public method? Renaming it is equivalent in a sense that devs have to change their code.

Reply:
The 1.4.6 name was `write_scratch_to_db`. Currently on `dev`, that function is now called `write_scratch_to_lmdb`, so from the users' point of view, `write_scratch_to_db` doesn't exist anymore. The proposed change reinstated `write_scratch_to_db` as deprecated, but now with @Fraser999 's suggestion, we just reverted the rename and avoided the breaking change.